File: src/main/java/com/study/platform/domain/comment/model/CommentRepository.java
package com.study.platform.domain.comment.model;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Optional;
import java.util.UUID;

public interface CommentRepository {

    Comment save(Comment comment);
    void delete(Comment comment);
    Page<Comment> findAllByPostIdWithAuthor(UUID postId, Pageable pageable);
    Optional<Comment> findByIdWithAuthor(UUID commentId);
}

File: domain/src/main/kotlin/com/study/platform/domain/comment/model/CommentRepository.kt
package com.study.platform.domain.comment.model

import java.util.UUID

interface CommentRepository {

    fun save(comment: Comment): Comment
    fun delete(comment: Comment)
    fun deleteAllByPostId(postId: UUID)
    fun findById(commentId: UUID): Comment?
    fun findByIdWithAuthor(commentId: UUID): Comment?
}
